service-workers/service-worker/resources/test-request-headers-worker.py

```diff
@@ -1,18 +1,21 @@
 import json
 import os
 import uuid
+import sys
+
+from wptserve.utils import isomorphic_decode
 
 def main(request, response):
-  path = os.path.join(os.path.dirname(__file__),
-                      "test-request-headers-worker.js")
-  body = open(path, "rb").read()
+  path = os.path.join(os.path.dirname(isomorphic_decode(__file__)),
+                      u"test-request-headers-worker.js")
+  body = open(path, u"rb").read()
 
-  data = {key:request.headers[key] for key,value in request.headers.iteritems()}
-  body = body.replace("%HEADERS%", json.dumps(data))
-  body = body.replace("%UUID%", str(uuid.uuid4()))
+  data = {isomorphic_decode(key):isomorphic_decode(request.headers[key]) for key, value in request.headers.items()}
+  body = body.replace(b"%HEADERS%", json.dumps(data).encode(sys.getdefaultencoding()))
+  body = body.replace(b"%UUID%", str(uuid.uuid4()).encode(sys.getdefaultencoding()))
```

Comment:
I'm not sure why you wanted to use `sys.getdefaultencoding()` here.

Both `json.dumps` and `str(uuid.uuid4())` always produce valid Unicode text strings. We should be able to safely call `encode(utf-8)` on them regardless of the default system encoding. Besides, we could simply perhaps read `body` as text (`"rt"` mode) and not worry about encoding.